Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al

```diff
@@ -591,6 +604,41 @@ codeunit 4022 "GP Vendor Migrator"
         until GPSY06000.Next() = 0;
     end;
 
+    local procedure GetBankAccountCode(VendorNo: Code[20]; var GPSY06000: Record "GP SY06000"; var BankAccountCounter: Integer): Code[20]
+    var
+        BankCode: Code[20];
+    begin
+        // If the bank account has a code, try to use it
+        BankCode := CopyStr(GPSY06000.EFTBankCode.Trim(), 1, MaxStrLen(BankCode));
+        if BankCode <> '' then
+            if not BankAccountAlreadyExists(VendorNo, BankCode) then
+                exit(BankCode);
+
+        // If the bank account code is blank (or already exists), default to the Vendor No.
+        BankCode := VendorNo;
+        if GPSY06000.Count() = 1 then
+            if not BankAccountAlreadyExists(VendorNo, BankCode) then
+                exit(BankCode);
+
+        // If the Vendor has more than one account, append a number to the code
```

Comment:
Not necessarily because of the first account. The counter would be 1, but what this is trying to determine is does the Vendor have multiple accounts that will be created, and if so append the number. If not, don't append the number, just assign it the Vendor No. without a suffix.